third_party/nvfuser/csrc/python_frontend/fusion_record.h

```diff
@@ -836,7 +837,7 @@ struct CastOpRecord : RecordFunctor {
             _name,
             RecordType::CastOp),
         fusion_op_(fusion_op),
-        dtype_(dtype) {}
+        dtype_(std::get<Nvf::PrimDataType>(dtype.type)) {}
```

Comment:
It looks like the Python side is generally using the `PrimDataType` to expose data types.  Why did you leave the leave the `RecordFunctor` child class constructors passing in the `Nvf::DataType` instead of have them pass in the `Nvf::PrimDataType`?

Reply:
You are right. I changed `RecordFunctor` subclasses to use `PrimDataType`.